CMakeLists.txt

```diff
@@ -173,7 +176,7 @@ if(BUILD_TESTS)
 endif()
 
 target_include_directories(azure-storage-lite PUBLIC ${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} PRIVATE ${EXTRA_INCLUDE_DIRS})
-target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES})
+target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES} ${CMAKE_DL_LIBS})
```

Comment:
Do you need `${CMAKE_DL_LIBS}` here? I think the one `test/CMakeLists.txt` might be sufficient.